tests/unit/validateIdentity.test.ts

```diff
@@ -0,0 +1,90 @@
+import { HttpStatus } from "@fwl/web";
+import fetch, { enableFetchMocks } from "jest-fetch-mock";
+
+import {
+  InvalidValidationCode,
+  TemporaryIdentityExpired,
+} from "@src/clientErrors";
+import { validateIdentity } from "@src/components/display/fewlines/ValidateIdentityForm/ValidateIdentityForm";
+import * as fetchJson from "@src/utils/fetchJson";
+
+enableFetchMocks();
+
+describe("validateIdentity unit test", () => {
+  const mockedFetchJson = jest.spyOn(fetchJson, "fetchJson");
+
+  beforeAll(() => {
+    fetch.resetMocks();
+  });
+
+  it("should throw a new InvalidValidationCode error if validation code isn't valid", async (done) => {
+    expect.assertions(3);
+
+    const mockedFetchResponse = new Response(
+      JSON.stringify({ error: "INVALID" }),
+      {
+        status: HttpStatus.BAD_REQUEST,
+      },
+    );
+
+    mockedFetchJson.mockResolvedValueOnce(mockedFetchResponse);
+
+    const error = await validateIdentity("000000", "1234").catch((error) => {
+      return error;
+    });
+
+    expect(mockedFetchJson).toHaveBeenCalled();
+    expect(error).toBeInstanceOf(InvalidValidationCode);
+    expect(error).not.toBeInstanceOf(TemporaryIdentityExpired);
+
+    done();
```

Comment:
I think we need to talk about how the jest callback is dealt with ^^.

If the callback is `async`, you will (probably) never need the `done()` call because there's three cases for the jest callback with asynchronous code:

```js
test("not working async code", () => {
  expect.assertions(1)
  fetch(url)
    .then(() => {
      expect(1).toBe(1);
    })
  // this will not work since the test will end before an expect call has been made
})

test("working async code with done callback", (done) => {
  expect.assertions(1)
  fetch(url)
    .then(() => {
      expect(1).toBe(1);
      done()
    })
})

test("working with returning a promise", () => {
  expect.assertions(1)
  return fetch(url)
    .then(() => {
      expect(1).toBe(1);
    })
})
```

And declaring your function as `async` will make it return a Promise, even if it's `Promise<void>` 

Please tell me if it's not super clear, I would love to have everyone in the team be sure of what they do in those kind of tests 🙏

Reply:
Hmm nope it's not super clear in my mind indeed, I think I need to read a bit more of documentation concerning this subject to make things clearer with `async` callback.